guardrails/llm_providers.py

```diff
@@ -364,6 +484,31 @@ def get_llm_ask(llm_api: Callable, *args, **kwargs) -> PromptCallableBase:
     except ImportError:
         pass
 
+    try:
+        from transformers import (  # noqa: F401 # type: ignore
+            FlaxPreTrainedModel,
+            PreTrainedModel,
+            TFPreTrainedModel,
+        )
+
+        api_self = getattr(llm_api, "__self__", None)
+
+        if (
+            isinstance(api_self, PreTrainedModel)
+            or isinstance(api_self, TFPreTrainedModel)
+            or isinstance(api_self, FlaxPreTrainedModel)
+        ):
+            return HuggingFaceModelCallable(*args, model_generate=llm_api, **kwargs)
+    except ImportError:
+        pass
+    try:
+        from transformers import Pipeline  # noqa: F401 # type: ignore
+
+        if isinstance(llm_api, Pipeline):
```

Comment:
Could we add an extra check here where we specifally check for a "text-generation" pipeline? This would handle cases wherein the users create and pass in a non-text-generation pipeline e.g. a text-classification pipeline.